src/org/traccar/database/DataManager.java

```diff
@@ -334,11 +346,34 @@ public void removeUser(long userId) throws SQLException {
                 .executeQuery(GroupPermission.class);
     }
 
-    public Collection<Device> getAllDevices() throws SQLException {
+    private Collection<Device> getAllDevices() throws SQLException {
         return QueryBuilder.create(dataSource, getQuery("database.selectDevicesAll"))
                 .executeQuery(Device.class);
     }
 
+    public Collection<Device> getAllDevicesCached() {
```

Comment:
`getAllDevices` were used in three places `GeofenceManager`, `api/resource/DeviceResource.java`
and `PermissionsManager` and return new collection of objects just created from db.
In first two places we do not need new clean objects without `geofencesIds` we need already filled objects with `geofenceIds`. They are already filled in `Map<> devicesById` and keep up to date by `GeofenceEventHandler`. Because of that we need function that return collection of cached devices `getAllDevicesCached()`, `getDeviceById` and `getDeviceByUniqueId` can't return whole collection to iterate or send with API.